src/Sylius/Behat/Context/Api/Admin/ManagingAdministratorsContext.php

```diff
@@ -129,13 +155,41 @@ public function iDeleteAdministratorWithEmail(AdminUserInterface $adminUser): vo
     }
 
     /**
+     * @Given /^I am editing (my) details$/
      * @When /^I want to edit (this administrator)$/
      */
     public function iWantToEditThisAdministrator(AdminUserInterface $adminUser): void
     {
         $this->client->buildUpdateRequest((string) $adminUser->getId());
     }
 
+    /**
+     * @When /^I (?:|upload|update) the "([^"]+)" image as (my) avatar$/
+     */
+    public function iUploadTheImageAsMyAvatar(string $avatar, AdminUserInterface $administrator): void
+    {
+        $response = $this->client->upload(
+            '/new-api/avatar-images',
+            ['owner' => $this->iriConverter->getIriFromItem($administrator)],
+            ['file' => new UploadedFile($this->minkParameters['files_path'] . $avatar, basename($avatar))]
```

Comment:
I'm wondering, is it a proper approach 🤔 It's not resolving the issue like [this one](https://github.com/Sylius/Sylius/issues/8864) when one wants to just use the API (not call the endpoint in the code). Maybe we should just pass the URL to the image which is temporarily somewhere else? I don' know, just thinking out loud 😄

Reply:
Probably I don't understand you in 100%, the case is to upload an image, not to pass a URL to the image... I have no other idea how to approach this.

Reply:
If I understood correctly, there is an ongoing problem in Sylius documentation on how to handle files upload through API since the Admin API. Problem is that what you have used here (and is used in the whole Sylius codebase) works because we are operating inside of a project. One could not simply use `UploadedFile` in js.